lib/moleculer/service/base.rb

```diff
@@ -74,8 +85,27 @@ def events
           @events ||= {}
         end
 
+        ##
+        # returns the full name of the service, including version and prefix
+        def full_name
+          return @full_name if @full_name
+
+          name    = service_name.dup
+          version = @version
+          version.prepend("v") if @version.is_a? Numeric
```

Comment:
would it make more sense here to ensure that the version is validated and fail fast for the services instead of having to check here?